image-hd.c

```diff
@@ -508,6 +538,19 @@ static int check_overlap(struct image *image, struct partition *p)
 		if (q->offset >= p->offset + p->size)
 			continue;
 
+		/*
+		 * Or maybe the image occupying the q partition has an
+		 * area which it is ok to overwrite. We do not do the
+		 * "vice versa" check, since images are written to the
+		 * output file in the order the partitions are
+		 * specified.
+		 */
+		start = max_ull(p->offset, q->offset);
+		end = min_ull(p->offset + p->size, q->offset + q->size);
+
+		if (image_has_hole_covering(q->image, start - q->offset, end - q->offset))
+			continue;
+
 		image_error(image,
 			    "partition %s (offset 0x%llx, size 0x%llx) overlaps previous "
 			    "partition %s (offset 0x%llx, size 0x%llx)\n",
```

Comment:
This check will break existing configs. We need to document the hole syntax and add some kind off comment here, so that users know how to fix this.

Reply:
Can you share some config that will break, maybe boil it down to a test case? Or did I capture the essence of what such a config looks like in the tests I added?

Reply:
Your test looks mostly good for that, but for some SoCs there is special stuff at 1k Offset, so in that case the hole ends there. That's why we need the `gpt-location` in this case.

Reply:
The last test case I add is specifically targeted at a Wandboard (I guess other imx boards behave similarly). But that case "just" needs to move the gpt array, it doesn't have anything that must sit at offset 0 and extend across the MBR table. But I guess one could have a vendor-provided bootloader image that is meant to be written at offset 0 and has garbage in the first 1K.